## The goal of this python notebook is to (1) recreate datasheet 4 from Daniels et. al (2015) using our data, (2) make direct comparisons between the two outputs, and (3) discuss issues with reproducibility that we ran in to during this project. 

In [2]:
import pandas as pd

In [4]:
# Read in the csv files as panda dataframe. 

coral = pd.read_csv('coral-Copy1.tab', sep = '\t', header=None, names=['contig', 'e-value', 'bitscore']) # this is the data sheet that contains the coral assigned contigs only. 
algae = pd.read_csv('algae-Copy1.tab', sep = '\t', header=None, names=['contig', 'e-value', 'bitscore']) # this is the data sheet that contains the algae assigned contigs only. 
bngm = pd.read_csv('BLASTX_and_GO_merged.tab', sep = '\t') # this is the file that contains the merged blastx and go annotations. 
pval = pd.read_csv("DEgenes_pval0.1_coral_filttxm.tsv", sep = '\t', names=['contig', "logFC", 'logCPM', 'p-value', 'FDR']) # this reads in the file that had a p-value cutoff of 0.01 
allval = pd.read_csv("DEgenes_coral_filttxm.tsv", sep = '\t', names=['contig', "logFC", 'logCPM', 'p-value', 'FDR']) # this reads in the file that had no p-value cut off
daniels = pd.read_csv("DataSheet4_Danielsetal2015.txt", sep = '\t') # this reads in the data sheet from Daniels et. al that we are going to compare our resutls to. 

In [7]:
coral['Name']='coral' # add a column called Name to the coral object and each row will identify that contig as coral. 
coral.head(n=5)

contig       e-value  bitscore   Name
0  TRINITY_DN10006_c0_g1_i12  4.400000e-42     176.0  coral
1   TRINITY_DN10006_c0_g1_i5  3.400000e-43     178.0  coral
2  TRINITY_DN100511_c0_g1_i1  0.000000e+00     953.0  coral
3   TRINITY_DN10129_c0_g1_i3  0.000000e+00    1075.0  coral
4   TRINITY_DN10144_c0_g1_i1  1.170000e-97     359.0  coral

In [8]:
algae['Name']='algae' # add a column called Name to the algae object and each row read algae to identify that contig as algae. 
algae.head(n=5)

contig       e-value  bitscore   Name
0   TRINITY_DN10006_c0_g1_i6  8.030000e-46     187.0  algae
1   TRINITY_DN10006_c0_g1_i8  8.040000e-46     187.0  algae
2   TRINITY_DN10006_c0_g2_i1  9.290000e-99     363.0  algae
3   TRINITY_DN10006_c1_g1_i1  4.430000e-96     353.0  algae
4  TRINITY_DN100150_c0_g1_i1  0.000000e+00    1312.0  algae

In [9]:
result = coral.append(algae) 
# this creates a new data table called result, and appends the algae table to the bottom of the coral table. 

In [38]:
result2 = pval.merge(result, how="left", left_on="contig", right_on="contig")

# this creates a new data table called result2. 
#This is a dataframe that combines the result table to the pval table. 

In [18]:
result3 = result2.merge(bngm, how='left', left_on="contig", right_on="contig")
# this creates a new data table called result3. 
#This is a dataframe that combines the previous table and the bngm table. 

result3.loc[result3['Name']== 'algae'] ## prints the rows where the Name column contains the word algae. 

contig     logFC    logCPM   p-value  FDR  e-value  \
704   TRINITY_DN46548_c0_g1_i1 -4.386382  6.045601  0.068732    1      0.0   
791   TRINITY_DN12201_c0_g1_i1 -4.386411  6.045601  0.069566    1      0.0   
1008   TRINITY_DN1883_c0_g2_i1  4.398525  6.046961  0.081724    1      0.0   

      bitscore_x   Name  Unnamed: 0   db  ...    Status  \
704       2401.0  algae     39366.0   sp  ...  reviewed   
791       1024.0  algae         NaN  NaN  ...       NaN   
1008      1471.0  algae         NaN  NaN  ...       NaN   

                                          Protein names  Gene names  \
704   Beta-glucanase (EC 3.2.1.73) (1,3-1,4-beta-D-g...   licB lam1   
791                                                 NaN         NaN   
1008                                                NaN         NaN   

                                               Organism  Length  \
704   Hungateiclostridium thermocellum (Clostridium ...   334.0   
791                                                 NaN     NaN   
1008                                                NaN     NaN   

                 Gene ontology (biological process)  \
704   polysaccharide catabolic process [GO:0000272]   
791                                             NaN   
1008                                            NaN   

      Gene ontology (cellular component)  \
704                                  NaN   
791                                  NaN   
1008                                 NaN   

                                     Gene ontology (GO)  \
704   licheninase activity [GO:0042972]; polysacchar...   
791                                                 NaN   
1008                                                NaN   

      Gene ontology (molecular function)       Gene ontology IDs  
704    licheninase activity [GO:0042972]  GO:0000272; GO:0042972  
791                                  NaN                     NaN  
1008                                 NaN                     NaN  

[3 rows x 34 columns]

## Notice that only one of these differentiall expressed genes are annotated.  

In [19]:
result3.loc[result3['Name']== 'coral'] ## prints the rows where the 'Name' column contains the word coral. 

contig     logFC    logCPM   p-value  FDR  e-value  \
379   TRINITY_DN2009_c0_g1_i1  4.650113  6.154022  0.043862    1      0.0   
654  TRINITY_DN14168_c0_g2_i1 -2.711862  6.432680  0.059627    1      0.0   

     bitscore_x   Name  Unnamed: 0   db  ... Status Protein names  Gene names  \
379      1951.0  coral         NaN  NaN  ...    NaN           NaN         NaN   
654      1705.0  coral         NaN  NaN  ...    NaN           NaN         NaN   

     Organism  Length  Gene ontology (biological process)  \
379       NaN     NaN                                 NaN   
654       NaN     NaN                                 NaN   

     Gene ontology (cellular component)  Gene ontology (GO)  \
379                                 NaN                 NaN   
654                                 NaN                 NaN   

     Gene ontology (molecular function)  Gene ontology IDs  
379                                 NaN                NaN  
654                                 NaN                NaN  

[2 rows x 34 columns]

## Notice that none of these differentially expressed coral genes are annotated. 

In [20]:
result3.to_csv("DataSheet4Replica.tab" , sep = '\t')# Write out a tab delimited file containing the merged information using the 0.01 p-val cutoff data

## In the above results, we only had two differentially expressed coral contigs, and three differentially expressed algae contigs. We decided to recombine the files using the edgeR output with no p-value cutoff. 

In [26]:
allresult2 = allval.merge(result, how="left", left_on="contig", right_on="contig")

In [29]:
allresult3 = allresult2.merge(bngm, how='left', left_on="contig", right_on="contig")
# this creates a new data table called result3. 
#This is a dataframe that combines the previous table and the bngm table. 

allresult3.loc[allresult3['Name']== 'coral'] ## prints the rows where the Name column contains the word coral 

contig                logFC            logCPM  \
380      TRINITY_DN2009_c0_g1_i1     4.65011343019195  6.15402244420148   
655     TRINITY_DN14168_c0_g2_i1     -2.7118615628497  6.43268015511166   
1157     TRINITY_DN8455_c0_g2_i1    -1.64513392793553  6.73758089842268   
1211   TRINITY_DN110960_c0_g1_i1    -4.38947751282279  6.04628124884687   
1344     TRINITY_DN1038_c0_g1_i3    -4.38277831733629  6.04424119956887   
...                          ...                  ...               ...   
39841     TRINITY_DN135_c0_g2_i3  0.00948505727106399  5.93133057425655   
40483     TRINITY_DN338_c0_g1_i1  0.00686972109822287  5.93133057425655   
40918    TRINITY_DN6741_c1_g1_i1  0.00666592596356015  5.93133057425655   
41127    TRINITY_DN4941_c1_g1_i1  0.00562822783377648  6.15402244420148   
41191    TRINITY_DN7156_c0_g2_i1  0.00530049844155792  6.34693767785712   

                  p-value  FDR        e-value  bitscore_x   Name  Unnamed: 0  \
380    0.0438620329057715  1.0   0.000000e+00      1951.0  coral         NaN   
655    0.0596267087933166  1.0   0.000000e+00      1705.0  coral         NaN   
1157    0.108752511632866  1.0   0.000000e+00      2396.0  coral     50728.0   
1211    0.110889992626927  1.0  1.160000e-171       604.0  coral     60232.0   
1344    0.114242753621899  1.0   0.000000e+00      4324.0  coral     43071.0   
...                   ...  ...            ...         ...    ...         ...   
39841                   1  1.0   0.000000e+00       723.0  coral         NaN   
40483                   1  1.0   0.000000e+00      2700.0  coral         NaN   
40918                   1  1.0  2.900000e-135       483.0  coral     12140.0   
41127                   1  1.0   0.000000e+00       715.0  coral         NaN   
41191                   1  1.0   0.000000e+00      2030.0  coral     48104.0   

        db  ...    Status                                      Protein names  \
380    NaN  ...       NaN                                                NaN   
655    NaN  ...       NaN                                                NaN   
1157    sp  ...  reviewed                  Tubulin beta chain (Beta-tubulin)   
1211    sp  ...  reviewed                        60S ribosomal protein L12-B   
1344    sp  ...  reviewed  Pyruvate, phosphate dikinase (EC 2.7.9.1) (Pyr...   
...    ...  ...       ...                                                ...   
39841  NaN  ...       NaN                                                NaN   
40483  NaN  ...       NaN                                                NaN   
40918   sp  ...  reviewed  Tubulin alpha-1 chain [Cleaved into: Detyrosin...   
41127  NaN  ...       NaN                                                NaN   
41191   sp  ...  reviewed  Probable 3-beta-hydroxysteroid-Delta(8),Delta(...   

                                              Gene names  \
380                                                  NaN   
655                                                  NaN   
1157                                                 NaN   
1211                               rpl1202 SPCC31H12.04c   
1344                           ppdK podA R00932 SMc00025   
...                                                  ...   
39841                                                NaN   
40483                                                NaN   
40918                                                NaN   
41127                                                NaN   
41191  Os01g0103600 LOC_Os01g01369 P0436E04.5 P0455C0...   

                                                Organism  Length  \
380                                                  NaN     NaN   
655                                                  NaN     NaN   
1157                                    Sus scrofa (Pig)   445.0   
1211   Schizosaccharomyces pombe (strain 972 / ATCC 2...   165.0   
1344   Rhizobium meliloti (strain 1021) (Ensifer meli...   898.0   
...                                                  ...     ...   
39841 

In [28]:
allresult3.loc[allresult3['Name']== 'algae']  ## prints the rows where the Name column contains the word algae

contig                logFC            logCPM  \
705     TRINITY_DN46548_c0_g1_i1    -4.38638220717413  6.04560113160882   
792     TRINITY_DN12201_c0_g1_i1    -4.38641119181608  6.04560113160882   
1009     TRINITY_DN1883_c0_g2_i1     4.39852510924497  6.04696146675522   
1196    TRINITY_DN15408_c0_g1_i2    -4.38947408265821  6.04628124884687   
1480    TRINITY_DN82260_c0_g1_i1    -4.38979722546205  6.04628124884687   
...                          ...                  ...               ...   
40205   TRINITY_DN20524_c0_g1_i1  0.00693817676603883  5.93133057425655   
40399  TRINITY_DN152705_c0_g1_i1  0.00689805237482067  5.93133057425655   
40626     TRINITY_DN513_c0_g2_i2  0.00681616174201276  5.93133057425655   
40937   TRINITY_DN79727_c0_g1_i1  0.00666080958696821  5.93133057425655   
40940       TRINITY_DN7_c0_g3_i1  0.00665898380066986  5.93133057425655   

                  p-value  FDR        e-value  bitscore_x   Name  Unnamed: 0  \
705    0.0687317448930154  1.0   0.000000e+00      2401.0  algae     39366.0   
792    0.0695657589360658  1.0   0.000000e+00      1024.0  algae         NaN   
1009   0.0817235723286364  1.0   0.000000e+00      1471.0  algae         NaN   
1196    0.110800601488179  1.0  3.060000e-130       466.0  algae         NaN   
1480    0.119896285209172  1.0   0.000000e+00      1404.0  algae     44470.0   
...                   ...  ...            ...         ...    ...         ...   
40205                   1  1.0  3.910000e-100       364.0  algae     21292.0   
40399                   1  1.0   0.000000e+00      1945.0  algae     51915.0   
40626                   1  1.0   0.000000e+00      1482.0  algae         NaN   
40937                   1  1.0  2.320000e-122       438.0  algae         NaN   
40940                   1  1.0   3.700000e-41       169.0  algae         NaN   

        db  ...    Status                                      Protein names  \
705     sp  ...  reviewed  Beta-glucanase (EC 3.2.1.73) (1,3-1,4-beta-D-g...   
792    NaN  ...       NaN                                                NaN   
1009   NaN  ...       NaN                                                NaN   
1196   NaN  ...       NaN                                                NaN   
1480    sp  ...  reviewed  Tetrapyrrole-binding protein, chloroplastic (G...   
...    ...  ...       ...                                                ...   
40205   sp  ...  reviewed                     Heat shock-like 85 kDa protein   
40399   sp  ...  reviewed  Succinate--CoA ligase [ADP-forming] subunit al...   
40626  NaN  ...       NaN                                                NaN   
40937  NaN  ...       NaN                                                NaN   
40940  NaN  ...       NaN                                                NaN   

                      Gene names  \
705                    licB lam1   
792                          NaN   
1009                         NaN   
1196                         NaN   
1480   GUN4 At3g59400 F25L23_260   
...                          ...   
40205                        NaN   
40399         At5g08300 F8L15_30   
40626                        NaN   
40937                        NaN   
40940                        NaN   

                                                Organism  Length  \
705    Hungateiclostridium thermocellum (Clostridium ...   334.0   
792                                                  NaN     NaN   
1009                                                 NaN     NaN   
1196                                                 NaN     NaN   
1480              Arabidopsis thaliana (Mouse-ear cress)   265.0   
...                                                  ...     ...   
40205                                  Trypanosoma cruzi   704.0   
40399             Arabidopsis thaliana (Mouse-ear cress)   347.0   
40626                                                NaN     NaN   
40937                                                NaN     NaN   
40940                 

In [30]:
allresult3.to_csv("DataSheet4Replica-all-p-val.tab" , sep = '\t')# Write out a tab delimited file containing the merged information from the file with no p-value cut off

## Now we will compare our results to Daniels et. al (2015). 

In [37]:
#Drop rows which contains NaN or missing value in the "Name" and "Gene ontology (GO)" columns. 
modResult3 = result3.dropna(how='any', subset=['Name', 'Gene ontology (GO)']) 
 
print("Contents of the Modified Dataframe : ")
modResult3.head(n=3)

Contents of the Modified Dataframe : 


contig     logFC    logCPM   p-value  FDR  e-value  \
704  TRINITY_DN46548_c0_g1_i1 -4.386382  6.045601  0.068732    1      0.0   

     bitscore_x   Name  Unnamed: 0  db  ...    Status  \
704      2401.0  algae     39366.0  sp  ...  reviewed   

                                         Protein names  Gene names  \
704  Beta-glucanase (EC 3.2.1.73) (1,3-1,4-beta-D-g...   licB lam1   

                                              Organism  Length  \
704  Hungateiclostridium thermocellum (Clostridium ...   334.0   

                Gene ontology (biological process)  \
704  polysaccharide catabolic process [GO:0000272]   

     Gene ontology (cellular component)  \
704                                 NaN   

                                    Gene ontology (GO)  \
704  licheninase activity [GO:0042972]; polysacchar...   

     Gene ontology (molecular function)       Gene ontology IDs  
704   licheninase activity [GO:0042972]  GO:0000272; GO:0042972  

[1 rows x 34 columns]

In [57]:
modResult3

contig     logFC    logCPM   p-value  FDR  e-value  \
704  TRINITY_DN46548_c0_g1_i1 -4.386382  6.045601  0.068732    1      0.0   

     bitscore_x   Name  Unnamed: 0  db  ...    Status  \
704      2401.0  algae     39366.0  sp  ...  reviewed   

                                         Protein names  Gene names  \
704  Beta-glucanase (EC 3.2.1.73) (1,3-1,4-beta-D-g...   licB lam1   

                                              Organism  Length  \
704  Hungateiclostridium thermocellum (Clostridium ...   334.0   

                Gene ontology (biological process)  \
704  polysaccharide catabolic process [GO:0000272]   

     Gene ontology (cellular component)  \
704                                 NaN   

                                    Gene ontology (GO)  \
704  licheninase activity [GO:0042972]; polysacchar...   

     Gene ontology (molecular function)       Gene ontology IDs  
704   licheninase activity [GO:0042972]  GO:0000272; GO:0042972  

[1 rows x 34 columns]

## Our output resulted in only one differentially expressed gene that was assigned to an organism, in this case algae, and annotated. 
The single gene that we had was down-regulated in diseased coral relative to healthy coral. The false discovery rate is 1, despite all of our efforts to analyze the data and follow Carolyn's suggestions. While this doesn't pass the parameters used in the paper, we decided to do a standard p-value cutoff of 0.1. This gene involved in polysaccharide catabolic processes. 

## Daniels et. al (2015) had 308 DE genes for coral and 51 for symbiodinium (algae). Of these DE genes, 167 coral genes were anntotated, and 47 algae genes were annotated. 
Each of these genes had a false discovery rate of less than 0.1. Some of the shared genes between coral and algae were involved in polyubiquitin, oxidative stress, translation, and heat shock proteins. You can see a summary of all of the annotated genes in the table output below. 

In [55]:
modDaniels = daniels[daniels.organism != '-'] # this saves the dataframe as a new dataframe that does not 
    # include the rows that contained "-" which was equal to no data in the column organism. 
modDaniels2 = modDaniels.dropna(how='any', subset=['Name', 'organism']) # this saves the dataframe as a new dataframe 
# that does not include an NaN in the column name or organism. 


In [58]:
print("Daniels et.al annotated genes")
modDaniels2

Daniels et.al annotated genes


trinity_mixed                                         description  \
0    comp20584_c0_seq1                                           Stathmin   
1    comp24214_c0_seq1                        Radial spoke head 1 homolog   
2    comp15471_c0_seq1                                  Predicted protein   
3    comp30985_c0_seq1                                       Histone H1.0   
4    comp33223_c0_seq3                                  Predicted protein   
..                 ...                                                ...   
350  comp42074_c0_seq1                    60S acidic ribosomal protein P2   
351  comp44608_c0_seq1     N5-carboxyaminoimidazole ribonucleotide mutase   
352  comp15515_c0_seq1  Cytochrome b6-f complex iron-sulfur subunit 1,...   
353  comp45130_c0_seq1               UPF0051 protein ABCI8, chloroplastic   
354  comp45316_c0_seq1                          40S ribosomal protein S29   

         logFC    logCPM        PValue           FDR uniprot_id  \
0   -14.414809  8.838310  2.572130e-07  1.714050e-04     Q6DUB7   
1   -14.370998  6.366770  1.941690e-07  1.450830e-04     Q8WYR4   
2   -14.301689  5.743128  1.617640e-11  8.092000e-08     A7S0Y3   
3   -13.022110  7.445702  1.275200e-06  5.497080e-04     Q5NVN9   
4   -12.757571  4.752345  3.195880e-07  1.989970e-04     A7SMH1   
..         ...       ...           ...           ...        ...   
350  11.523608  3.642923  1.208710e-08  1.806290e-05     Q9UUZ6   
351  11.740401  3.925264  9.032560e-07  4.090380e-04     O58058   
352  12.323375  4.508718  2.931880e-07  1.877740e-04     Q5CC93   
353  12.679842  4.867161  3.831900e-07  2.260410e-04     Q9ZS97   
354  12.976470  5.132429  1.580820e-09  4.159980e-06     Q757G1   

                                              organism       evalue   Name  \
0                                           Sus scrofa  7.99424E-11  coral   
1                                         Homo sapiens  5.89905E-61  coral   
2                               Nematostella vectensis  7.31083E-47  coral   
3                                         Pongo abelii  3.34161E-20  coral   
4                               Nematostella vectensis   7.4874E-21  coral   
..                                                 ...          ...    ...   
350  Neosartorya fumigata (strain ATCC MYA-4609 / A...   1.9022E-08  algae   
351  Pyrococcus horikoshii (strain ATCC 700860 / DS...  1.17715E-51  algae   
352                                Cyanophora paradoxa  7.34038E-67  algae   
353                               Arabidopsis thaliana   3.415E-113  algae   
354  Ashbya gossypii (strain ATCC 10895 / CBS 109.5...  2.26586E-15  algae   

    length                                         Domains  \
0      149                  SLD (stathmin-like) domain (1)   
1      309                                MORN repeats (6)   
2      362                                               -   
3      194  H15 (linker histone H1/H5 globular) domain (1)   
4      755                                               -   
..     ...                                             ...   
350    111                                             NaN   
351    177                                               -   
352    239                               Rieske domain (1)   
353    557                                               -   
354     56                                               -   

                  Protein Families Unnamed: 13  Unnamed: 14  Unnamed: 15  \
0                  Stathmin family         NaN          NaN          NaN   
1                                -         NaN          NaN          NaN   
2                                -         NaN          NaN          NaN   
3             Histone H1/H5 family         NaN          NaN          NaN   
4                                -         NaN          NaN          NaN   
..                             ...         ...          ...          ...   
350  Ribosomal protein L12P family         NaN          NaN     


## Reproducibility ## 

Our results do not come close to matching the results from Daniels et. al (2015), and is a perfect example of why thinking about reproducibility is so important. One of the first issues that we ran into was when we retrieved the raw sequences from NCBI, the files had not been demultiplexed. We had to reach out to the author to request their original raw sequences. The main area that is likely the cause of reproducibility issues the step between quality trimming the reads with Trimmomatic and mapping the paired-end reads to the de novo transcriptome using Bowtie 2. Next we will outline some of the key steps that we made judgement calls on how to run specific steps that were not outlined in the methods section. 


### 1. Error Correction
Daniels et. al (2015) used **ALLPATHS-LG** for data error correction using a standalone module, however this is no longer supported, so we used a different package for error correction called **rCorrector** which is available for conda installation. There was no information included on the number of reads that were removed by **ALLPATHS-LG**, so it was not possible for us to assess how our error correction compared to theirs. 

### 2. Assembly with Trinity 
From Daniel's et. al (2015): "...assembled with the Trinity pipeline (version: r20131110) (Haas et al., 2013) to generate a de novo reference transcriptome (67,593 genes > 200 bp, mean length: 541 bp, N50: 656 bp) that contained assembled gene loci from coral, Symbiodinium, and other eukaryotes (e.g., endolithic algae, fungi, etc.)"

With no infromation on how they ran the Trinity pipeline, we used close to default settings to produce a de novo transcriptome. This generated a fasta file with **239,000 transcripts,  197,483 genes, mean length: 573bp, N50: 724bp**. 

We had no way of knowing how or whether or not they filtered their transcriptome to get the number of genes that they did, so we follow Carolyn's advice to filter out the longest isoform for each gene and then used **Salmon** psuedo-alignment against the longest-isoform transcriptome to get a TPM score, and then filtered based on TPM > 1 (expression based filtering). This generated a new transcriptome that has **66,044 genes**, much closer to the Daniels et.al **gene count of 67,593**; though we had a **average contig length of 740bp and a N50 of 932bp**.

### 3. Bowtie2 Alignment 
From Daniel's et. al (2015): "polyA+ libraries were mapped to the reference transcriptome using Bowtie2-2.1.0..."

There was no information provided about the parameters used for **Bowtie2**, so we used defualt settings to map our reads to the de novo transcriptome. Approximate 15% of our reads were mapped to exactly one contig, and over 70% were mapped to more than one contig. We do not know how this compared to Daniels et. al because no information was provided in the methods about this. 


The remainder of the steps in this pipeline went relatively smoothly and was well documented. 


### 4. Reference Transcriptome Annotation 
The next issue that we ran into was with compiling our custom cnidarian database. We had issues with finding the same genomes that they used, and then there was no explanation for how the custom database was compiled. We made the decision to use mostly default parameters.   

**Custom "Coral" Database:**
We were able to find the exact genomes for some of the coral species that they used. For the ones that we were unable to find the same genome. we chose to select the same species but different author. 

Acropora hyacinthus : A_hyacinth_GDIF01.1.fsa_nt.gz
We used the same transcriptome, which can be found at NCBI using the Barshis et al. 2013 paper at the link: https://www.ncbi.nlm.nih.gov/Traces/wgs/?val=GDIF01

The O. faveolata genome that we used is the same as used by Daniels et a. and was found at: https://www.ncbi.nlm.nih.gov/genome/annotation_euk/Orbicella_faveolata/100/  and named: GCF_002042975.1_ofav_dov_v1_rna.fna.gz

We used a different genome for S. pistillata because we were unable to find the referenced genome. We used the S. pistillata genome from: https://www.ncbi.nlm.nih.gov/assembly/GCF_002571385.1 and named it: GCF_002571385.1_Stylophora_pistillata_v1_rna.fna.gz

We downloaded the same Acropora digitifera genome as the paper from: https://marinegenomics.oist.jp/coral/viewer/download?project_id=3 and named it : adi_transcriptome_assembly.v1.fa.gz

**Custom"Other Cnidarians" in the coral database:**
The other cnidarians used in the paper - Aiptasia pallida, Hydra vulgaris, and Nematostella vectensis - were not specified in the paper methods, only referenced that these came from the NCBI database. We made the judgement call to select ones from the NCBI datebase. Additionally, the Aiptasia pallida genome that was used in Daniels et. al was unpublished, therefore we were unable to find exactly the same genome that they had used. 

H.vulgaris was downloaded from: https://www.ncbi.nlm.nih.gov/assembly/GCF_000004095.1/ and named: H.vulgaris_GCF_000004095.1_Hydra_RP_1.0_rna.fna.gz

N.vectensis was downloaded from: https://www.ncbi.nlm.nih.gov/assembly/GCF_000209225.1/ and named: N.vectensis_GCF_000209225.1_ASM20922v1_rna.fna.gz

A.pallida was downloaded from: https://www.ncbi.nlm.nih.gov/assembly/GCF_001417965.1/ and named: E.pallida_GCF_001417965.1_Aiptasia_genome_1.1_rna.fna.gz  Also note that this we could only find an RNA fasta for this species. 

**Custom Symbiodinium database** 
We did not run in to these same issues with the symbiodinium database. 
Symbidinium clade A & B were downloaded from: http://medinalab.org/zoox/
Symbiodinium clade C was downloaded from: https://www.ncbi.nlm.nih.gov/Traces/wgs/?val=GAFO01
Symbiodinium clade D was downloaded from: https://www.ncbi.nlm.nih.gov/Traces/wgs/?val=GAFP01
